src/main/java/network/brightspots/rcv/HartCvrReader.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Universal RCV Tabulator
+ * Copyright (c) 2017-2020 Bright Spots Developers.
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+ * the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this
+ * program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package network.brightspots.rcv;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import javafx.util.Pair;
+import network.brightspots.rcv.CastVoteRecord.CvrParseException;
+
+class HartCvrReader {
+
+  private final String cvrPath;
+  private final ContestConfig contestConfig;
+
+  HartCvrReader(String cvrPath, ContestConfig contestConfig) {
+    this.cvrPath = cvrPath;
+    this.contestConfig = contestConfig;
+  }
+
+  // iterate all xml files in the source input folder
+  void readCastVoteRecordsFromFolder(List<CastVoteRecord> castVoteRecords)
+      throws IOException, CvrParseException {
+    File cvrRoot = new File(this.cvrPath);
+    File[] children = cvrRoot.listFiles();
+    for (File child : children) {
+      if (child.getName().endsWith("xml")) {
+        readCastVoteRecord(castVoteRecords, child.toPath());
+      }
+    }
+  }
+
+  // parse Cvr xml file into CastVoteRecord objects and add them to the input List<CastVoteRecord>
+  void readCastVoteRecord(List<CastVoteRecord> castVoteRecords, Path path)
+      throws IOException, CvrParseException {
+    try {
+      Logger.log(Level.INFO, "Reading Hart cast vote record file: %s...", path.getFileName());
+
+      XmlMapper xmlMapper = new XmlMapper();
+      xmlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+      String xmlString = new String(Files.readAllBytes(path));
+      HartCvrXml xmlCvr = xmlMapper.readValue(xmlString, HartCvrXml.class);
+
+      for (Contest contest : xmlCvr.Contests) {
+        // TODO: use contest Id
```

Comment:
Are you going to update this in this PR? Or are you blocked on something?

Reply:
This is blocked on #456: support for contest ID.  As the ticket says, we could consider using contest name instead in which case the code here would not need to change.

Reply:
Oh, sorry, I thought maybe you had decided to handle #456 here. So I should still do #456. Got it.

Reply:
Please make this:
```
// TODO: use contest Id (blocked by #456)
```
And also add a comment in #456 with the text of this TODO included so we'll know to zoom back to it and address it.